crates/cubecl-opt/Cargo.toml

```diff
@@ -8,6 +8,9 @@ version.workspace = true
 [dependencies]
 cubecl-common = { path = "../cubecl-common", version = "0.2.0", default-features = false }
 cubecl-core = { path = "../cubecl-core", version = "0.2.0", default-features = false }
-num = { version = "0.4" }
+float-ord = "0.3"
+log = "0.4"
+num = "0.4"
 petgraph = { version = "0.6" }
+smallvec = { version = "1", features = ["union", "const_generics"] }
 stable-vec = { version = "0.4" }
```

Comment:
`smallvec` is an extremely small (heh) dependency (especially without anything like `serde` and it's used in an extremely hot loop. It's constructed multiple times for every instruction in the kernel, often thousands of times total. So keeping it on the stack in basically every case is a significant performance boost. Since it stores the arguments to an instruction, it can only go over its capacity for vectors > 4 (for OpenCL).

Reply:
As for putting it in the workspace, I just followed the idea that dependencies that are only used in one crate don't need to be in the workspace. But I'm not sure if that's actually the standard, I just saw some non-workspace dependencies in other crates.